### PR TITLE
Enhancement Message bus

### PR DESCRIPTION
**Pull request by MimiXav**
Just add Readme, and rename dto file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# fty-common-messagebus
+fty-common-messagebus:
+
+* Centralize all methods to adress Request/Reply, Publish/Subscribe patterns above malamute message bus.
```
Comment (boricj) [inline README.md]: adress -> address

File: include/fty_common_messagebus_dto.h [hunk 2]
```diff
@@ -19,10 +19,13 @@
...
 
-#ifndef FTY_COMMON_DTO_H_INCLUDED
-#define FTY_COMMON_DTO_H_INCLUDED
+#ifndef FTY_COMMON_MESSAGEBUS_DTO_H_INCLUDED
+#define FTY_COMMON_MESSAGEBUS_DTO_H_INCLUDED
 
 #include <string>
+#include <list>
+
+#include <fty_common_messagebus.h>
```
Comment (boricj) [inline include/fty_common_messagebus_dto.h]: Should be `#include "fty_common_messagebus_library.h"`

File: src/fty_common_messagebus_dto.cc [hunk 3]
```diff
@@ -32,5 +39,4 @@ void operator>> (messagebus::UserData &data, FooBar &object) {
     auto bar = data.front();
     data.pop_front();
     object = FooBar(foo, bar);
-}
-
+}
```
Comment (boricj) [inline src/fty_common_messagebus_dto.cc]: Put newline back.

**Review by boricj — APPROVED** [review]
Need to change include directive. Minor typos otherwise.

**Review by FrancoisRegisDegott-eaton — APPROVED** [review]

**Review by boricj — APPROVED** [review]

**Review by perrettecl — APPROVED** [review]
@MimiXav  Ok for me. Good job. Only concern (but I think it's just question of time) is the split between the interface and the "malamute" implementation. 

I would like to be able to make an implementation with  Unix Socket, and for that would be great to not have to link to any malamute code.

**Comment by perrettecl** [comment]
@MimiXav  and @boricj  there is also one thing that I don't get:

    MessageBus* connect(const std::string& endpoint, const std::string& clientName) {
        return new MessageBusMalamute(endpoint, clientName);
    }

In my point of you, this should not exist and we should have an interface "void connect() = 0" or something similar (or even nothing) which will allow us to use the following code:

// init the connection => Bus specific
MessageBus * myAwesomeBus = new MalamuteMessageBus( myEndpoint , myClientName);

or 

MessageBus * myAwesomeBus = new SerialMessageBus( myTTY, myTTYParity, myTTYSpeed, myClientName);

// Rest of exchanges are generic
myAwesomeBus.connect();
myAwesomeBus.sendRequest(...................);

**Review by boricj — APPROVED** [review]

**Review by boricj — APPROVED** [review]

**Pull request merged**